### PR TITLE
workbench: use simple/old file monitor events only

### PR DESCRIPTION
**Pull request by LarsGit223**
This removes the dependency/the requirement for GLib version 2.46.

### CODE REVIEW

File: workbench/src/wb_monitor.h [hunk 1]
```diff
@@ -22,7 +22,7 @@
 #include <glib.h>
 #include "wb_project.h"
 
-#if defined(HAVE_GIO) && GLIB_CHECK_VERSION (2, 46, 0)
+#if defined(HAVE_GIO)
```
Comment (codebrainz) [inline workbench/src/wb_monitor.h]: I don't think you need this `HAVE_GIO` stuff from the build system as GIO is part of GLib, Geany requires GIO already, and AFAIK some plugins too.

Comment (elextr) [reply workbench/src/wb_monitor.h]: Yes, Geany itself stopped checking back in [2011](https://github.com/geany/geany/commit/591417697010200441af6b6ed76ef28ef4731d83)

Comment (LarsGit223) [reply workbench/src/wb_monitor.h]: Yes, geany itself is not using it, I will remove it. Maybe it should be removed from other plugins also.

Comment (LarsGit223) [reply workbench/src/wb_monitor.h]: Done.

**Pull request merged**